### PR TITLE
Close #299 - Add re support to markdown codeblock

### LINKED ISSUE #299: re in markdown code block
Screen Shot 2019-06-09 at 10 14 30 PM

I was trying to add `re` into the markdown embedded code block for Reason but found it's occupied by `regex_python`. 

I am not sure if this is injected by some third-party extensions or built-in. But Is there anyway we can override this?

### PR DESCRIPTION
**Pull request by Huxpro**

### CODE REVIEW

File: editor-extensions/vscode/reason-markdown-codeblock.json [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
     }
   ],
   "repository": {
     "reason-code-block": {
-      "begin": "(reason|reasonml)(\\s+[^`~]*)?$",
-      "end": "(^|\\G)(?=\\s*[`~]{3,}\\s*$)",
+      "name": "markup.fenced_code.block.markdown",
+      "begin": "(^|\\G)(\\s*)(\\`{3,}|~{3,})\\s*(?i:(re|reason|reasonml)(\\s+[^`~]*)?$)",
+      "end": "(^|\\G)(\\2|\\s{0,3})(\\3)\\s*$",
+      "beginCaptures": {
```
Comment (jaredly) [inline editor-extensions/vscode/reason-markdown-codeblock.json]: Can you explain what all of these changes are doing? I would have expected that you'd only need to change line 11 to include `re|`

Comment (Huxpro) [reply editor-extensions/vscode/reason-markdown-codeblock.json]: @jaredly have you read my issues on vscode repo? https://github.com/microsoft/vscode/issues/75176
also this change: https://github.com/mjbvz/vscode-fenced-code-block-grammar-injection-example/commit/e333677e840fd28f382935650553736aaebc2207

**Comment by jaredly** [comment]
Cool, thanks for the context! in future, adding those references to the PR description will help things go faster :)

**Comment by Huxpro** [comment]
> Cool, thanks for the context! in future, adding those references to the PR description will help things go faster :)

Sure! I thought referring to #299 is sufficient so not directly mentioning at this PR.

**Pull request merged**